### PR TITLE
docs: added documentation on instrumentation

### PR DESCRIPTION
**Pull request by shahednasser**
Added new documentation chapter on instrumentation and observability

### CODE REVIEW

File: www/apps/book/app/debugging-and-testing/instrumentation/page.mdx [hunk 1]
```diff
@@ -0,0 +1,157 @@
+import { Prerequisites, TypeList } from "docs-ui"
+
+export const metadata = {
+  title: `${pageNumber} Configure Instrumentation`,
+}
+
+# {metadata.title}
+
+In this chapter, you'll about observability in Medusa and how to configure instrumentation with OpenTelemetry.
```
Comment (thetutlage) [inline www/apps/book/app/debugging-and-testing/instrumentation/page.mdx]: you'll learn about?

**Review by thetutlage — COMMENTED** [review]
Looks pretty good 💪

File: www/apps/book/app/debugging-and-testing/instrumentation/page.mdx (hunk 1 above)
Comment (shahednasser) [reply www/apps/book/app/debugging-and-testing/instrumentation/page.mdx]: Yes nice catch 😅

**Review by olivermrbl — APPROVED** [review]
Nice!

**Pull request merged**